### ISSUE TITLE
Incorrect requirements.txt in Magic Wand Example

### ISSUE THREAD

**Issue by laurenmurphyx64** [tensorflow/tensorflow#49473]
@tensorflow/micro

**System information**
- Host OS Platform and Distribution (e.g., Linux Ubuntu 16.04): **Linux Ubuntu 18.04**
- TensorFlow installed from (source or binary): **Source** 
- Tensorflow version (commit SHA if source): f26800a1e5b1199cfc1a5aca916edc836b541687
- TensorFlow Python package version: **2.4.0 (attempted)**
- Python version: **Python 3.8 64-bit**
- Pip version: **Pip 21.1.1**

**Describe the problem**

For the [TensorFlow Lite Micro Magic Wand example](https://github.com/tensorflow/tensorflow/tree/master/tensorflow/lite/micro/examples/magic_wand), attempting to install requirements following instructions in `README.md` fails.

```
Collecting numpy==1.16.2
  Downloading numpy-1.16.2.zip (5.1 MB)
     |████████████████████████████████| 5.1 MB 477 kB/s 
...
ERROR: Cannot install numpy==1.16.2 and tensorflow==2.4.0 because these package versions have conflicting dependencies.

The conflict is caused by:
    The user requested numpy==1.16.2
    tensorflow 2.4.0 depends on numpy~=1.19.2
```

Even when numpy is manually updated to 1.19.2 and the model is trained, (I think) tensorflow 2.4.0 doesn't recognize the operation:

```
...
Didn't find op for builtin opcode 'RESHAPE' version '1'
Failed to get registration from op code RESHAPE
Failed starting model allocation.
...
```

*Caveat*

To get the above result, I converted the Lite model with `xxd -i model.tflite > magic_wand_model_data.cc`, copied the model data and length from the TensorFlow tree into [the Zephyr tree's TensorFlow Magic Wand sample](https://github.com/zephyrproject-rtos/zephyr/tree/main/samples/modules/tensorflow/magic_wand) `magic_wand_model_data.cc`, and ran it on a Renode-emulated `litex_vexriscv` board.

Maybe there's a factor in that process that could've introduced an issue (and in that case, my apologies for filing this issue), but I also re-did that process after changing the `requirements.txt` back to the previous commit's `numpy==1.16.2` and `tensorflow==2.0.0-beta1`:

```
cd tensorflow/lite/micro/examples/magic_wand/train
python3.7 -m venv ./venv # following README.md
source venv/bin/activate
pip install --upgrade pip # if necessary
pip install -r requirements.txt
...
```

I didn't have the same issues during operation (although the model didn't recognize the sample inputs correctly this time I trained it - it's supposed to go RING CIRCLE RING CIRCLE):

```
*** Booting Zephyr OS build v2.6.0-rc1-310-g2a9b32d43f31  ***
Got accelerometer, label: accel-0
WING:
*         *         *
 *       * *       *
  *     *   *     *
   *   *     *   *
    * *       * *
     *         *
WING:
...
WING:
...
```

**Please provide the exact sequence of commands/steps when you ran into the problem**
```
cd tensorflow/lite/micro/examples/magic_wand/train
python3.8 -m venv ./venv
source venv/bin/activate
pip install --upgrade pip # if necessary
pip install -r requirements.txt
```

**Describe the expected behavior**

`pip install -r requirements.txt` should work and the scripts should work correctly with the TensorFlow version in the `requirements.txt`.

**Possible solutions**

- Downgrade requirements.txt `tensorflow==2.0.0-beta1`

OR

- Upgrade `numpy==1.19.2` to match `tensorflow==2.4.0`
- Rewrite scripts to accommodate TensorFlow 2.4.0 (?)

(This is my first time filing an issue on TensorFlow, so my apologies if this isn't a bug and please let me know if you'd like me to remove / add anything.)

**Comment by wangtz** [comment]
Hi Lauren,

Thanks for raising the issue.

I think the deps issues is related to [this](https://github.com/tensorflow/tflite-micro/pull/93/files) where the TF version is bumped to 2.5  I guess we can relax the version on numpy. For example 

```
numpy
```

instead of 
```
numpy==1.16.2
```

Do you mind help verify and raise a PR if possible? Thanks!



Regarding the model accuracy, if you're experiencing issues with the current model. Feel free to checkout the previous version, which might work better for you.

https://github.com/tensorflow/tensorflow/commit/59c06b9016700dbf1ab0cefc062d247345cdd0f0#diff-66dac78b23c096b0d1aad4338feef6d2233ae2579ca9f171f0c2ec78ffc80158



Moving forward, we have just released the next generation gesture recognition work, which is awesome.

You can find more information from this video: https://github.com/googlecreativelab/air-snare
https://github.com/googlecreativelab/tf4micro-motion-kit

For training using TFJS:
https://github.com/googlecreativelab/air-snare

For deployment on the device (Arduino)
https://github.com/googlecreativelab/tf4micro-motion-kit


It would be interesting to see if we can port this work on the board you're working on :-) The new work requirers BLE to collect training data and BLE to update the model on the fly.


Thanks,
Tiezhen

**Comment by mohantym** [comment]
Hi @laurenmurphyx64 !
We are checking to see whether you still need help in this issue . Have you checked [documents ](https://www.tensorflow.org/lite/microcontrollers)from latest versions(2.7)  yet? . Thanks!

**Comment by sachinprasadhs** [comment]
Now, in the `requirements.txt` file the version of `numpy` has been updated to `1.21.0` for the `tensorflow` version `2.5.3`, you can check the details [here](https://github.com/tensorflow/tflite-micro/blob/main/tensorflow/lite/micro/examples/magic_wand/train/requirements.txt). Thanks!
